plotutil/xyer.go

```diff
@@ -0,0 +1,82 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package plotutil
+
+import "github.com/gonum/plot/plotter"
+
+// xyer implements plotter.XYer for use by ZipXY.
+type xyer struct {
+	x, y []float64
+	len  int
+}
+
+// ZipXY is a convenience function to build a plotter.XYer from slices.
+// Length will be set to the shortest slice length.
+// Panics on empty or nil slices.
+func ZipXY(x, y []float64) plotter.XYer {
+	switch {
+	case len(x) == 0:
+		panic("x is an empty or nil slice")
+	case len(y) == 0:
+		panic("y is an empty or nil slice")
+	}
+	ln := len(x)
```

Comment:
I think @kortschak  said this earlier, but this should panic on unequal lengths.